AdminServer/appscale/admin/__init__.py

```diff
@@ -420,14 +458,144 @@ def post(self, project_id, service_id):
 class VersionHandler(BaseHandler):
   """ Manages particular service versions. """
 
-  def initialize(self, acc):
-    """ Defines an AppControllerClient.
+  def initialize(self, acc, ua_client, zk_client, version_update_lock,
+                 thread_pool):
+    """ Defines required resources to handle requests.
 
     Args:
       acc: An AppControllerClient.
+      ua_client: A UAClient.
+      zk_client: A KazooClient.
+      version_update_lock: A kazoo lock.
+      thread_pool: A ThreadPoolExecutor.
     """
     self.acc = acc
+    self.ua_client = ua_client
+    self.zk_client = zk_client
+    self.version_update_lock = version_update_lock
+    self.thread_pool = thread_pool
+
+  def get_version(self, project_id, service_id, version_id):
+    """ Fetches a version node.
+
+    Args:
+      project_id: A string specifying a project ID.
+      service_id: A string specifying a service ID.
+      version_id: A string specifying a version ID.
+    Returns:
+      A dictionary containing version details.
+    """
+    version_node = '/appscale/projects/{}/services/{}/versions/{}'.format(
+      project_id, service_id, version_id)
+
+    try:
+      version_json, _ = self.zk_client.get(version_node)
+    except NoNodeError:
+      raise CustomHTTPError(HTTPCodes.NOT_FOUND, message='Version not found')
 
+    return json.loads(version_json)
+
+  def version_from_payload(self):
+    """ Constructs version from payload.
+
+    Returns:
+      A dictionary containing version details.
+    """
+    update_mask = self.get_argument('updateMask', None)
+    if update_mask is None:
+      message = 'At least one field must be specified for this operation.'
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST, message=message)
+
+    desired_fields = update_mask.split(',')
+    supported_fields = {'appscaleExtensions.httpPort',
+                        'appscaleExtensions.httpsPort'}
+    for field in desired_fields:
+      if field not in supported_fields:
+        message = ('This operation is only supported on the following '
+                   'field(s): [{}]'.format(', '.join(supported_fields)))
+        raise CustomHTTPError(HTTPCodes.BAD_REQUEST, message=message)
+
+    try:
+      given_version = json_decode(self.request.body)
+    except ValueError:
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST,
+                            message='Payload must be valid JSON')
+
+    masked_version = utils.apply_mask_to_version(given_version, desired_fields)
```

Comment:
Ignoring fields is the purpose of the mask, so I don't think we need a warning. The use case is that a client can keep reusing the same version object without masking it client-side for each request.

GAE does not warn or give an error when parts of the version are ignored, and strangely it does not even warn if the masked object turns out to be empty.